### PR TITLE
[11.0][FIX] purchase_order_line_deep_sort: Fix sorted with string and False values

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa

### CODE REVIEW

File: purchase_order_line_deep_sort/models/purchase_order.py [hunk 1]
```diff
@@ -32,8 +32,12 @@ def _sort_purchase_line(self):
         def resolve_subfields(obj, line_order):
             subfields = line_order.split('.')
             res = obj
+            str_fields = (fields.Char, fields.Text)
             for subfield in subfields:
-                res = getattr(res, subfield)
+                if isinstance(res._fields[subfield], str_fields):
+                    res = getattr(res, subfield) or ''
+                else:
+                    res = getattr(res, subfield)
```
Comment (JayVora-SerpentCS) [inline purchase_order_line_deep_sort/models/purchase_order.py:40]: IMHO, only one line should work,
res = getattr(res, subfield) or ''
We would prefer to have the test case or example where you need If condition (may be a good comment on code)

Comment (sergio-teruel) [reply purchase_order_line_deep_sort/models/purchase_order.py:40]: Mhuuu!! But or '' os not always, if you order by a boolean field you do not want an empty string.
I will do tests for this use case... :smile:

**Comment by pedrobaeza** [comment]
Waiting a test case

**Comment by sergio-teruel** [comment]
Hi, I will try resume this PR:
The python function sorted crash when you try make a order string values with boolean values:
my_list = ['a', 'b', False, '2019-06-15'] --> TypeError: '<' not supported between instances of 'str' and 'int'
But NOT crash when the type list elements are numeric and boolean values
my_list = [1, 2, False] --> [False, 1, 2]
So when we make purchase order lines sorted method check column types form fields to set empty string for ('text', 'varchar', 'timestamp', 'date') column types.

Test case done.

File: purchase_order_line_deep_sort/models/purchase_order.py (hunk 1 above)
Comment (sergio-teruel) [reply purchase_order_line_deep_sort/models/purchase_order.py:40]: @JayVora-SerpentCS I change isinstance function by fields.column_type method.

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@JayVora-SerpentCS please review

**Review by JayVora-SerpentCS — APPROVED** [review]

**Pull request merged**